File: python/bob/bob.py
import string

def response(phrase):
    phrase = phrase.strip()
    nopunc = "".join([x for x in phrase if x not in string.punctuation])

    if len(phrase) == 0:
        return "Fine. Be that way!"
        
    elif nopunc.isupper() and phrase.endswith("?") is True:
        return "Calm down, I know what I'm doing!"
        
    elif nopunc.isupper() is True:
        return "Whoa, chill out!"
        
    elif phrase.endswith("?"):
        return "Sure."

    else:
        return "Whatever."

File: typescript/bob/bob.ts
export function hey(message: string): string {
  let isQuestion: boolean = /\?[ ]*$/u.test(message);
  let isYelled: boolean =
    !!message && /[A-Z]+/u.test(message) && !/[a-z]+/u.test(message);
  let isSilence: boolean = message.trim() === '';

  switch (true) {
    case isQuestion && isYelled:
      return "Calm down, I know what I'm doing!";
    case isQuestion:
      return 'Sure.';
    case isYelled:
      return 'Whoa, chill out!';
    case isSilence:
      return 'Fine. Be that way!';
    default:
      return 'Whatever.';
  }
}
